flask-api/flaskr/vote.py

```diff
@@ -0,0 +1,189 @@
+
+    db = get_db()
+
+    # Validate v_event_id
+    cursor = db.cursor()
+    v_event = cursor.execute(
+        'SELECT * FROM voting_event WHERE id = ?',
+        (v_event_id,)
+    ).fetchone()
+
+    if v_event is None:
+        abort(400, 'Bad Request')
+
+    # Check if user has voted
+    cursor = db.cursor()
+    v_status = cursor.execute(
+        'SELECT * FROM vote_status WHERE v_event_id = ? AND user_id = ?',
+        (v_event_id, user_id)
+    ).fetchone()
+
+    if not v_status is None:
+        abort(400, 'Bad Request')
+
+    # Validate vote data
+    cursor = db.cursor()
+    parties = cursor.execute(
+        'SELECT * FROM party WHERE v_event_id = ?',
+        (v_event_id,)
+    ).fetchall()
+    party_id_list = list(map(lambda party: party['id'], parties))
+
+    cursor = db.cursor()
+    candidates = cursor.execute(
+        'SELECT * FROM candidate WHERE v_event_id = ?',
+        (v_event_id,)
+    ).fetchall()
+    candidates_id_list = list(map(lambda candidate: candidate['id'], candidates))
+
+    print("TESTING POINT")
+
+    try:
+        atl_vote = vote_data['above']
+        btl_vote = vote_data['below']
+
+        # Check votes
+        if len(atl_vote) >= 6:
+            # Validate above the line votes
+            # Check if parties exist
+            for party in atl_vote:
+                if not party['party_id'] in party_id_list:
+                    abort(400, 'Bad Request')
+
+            # Check for no duplicate values
+            initial_length = len(atl_vote)
+            vote_values = list(map(lambda vote: vote['number'], atl_vote))
+            vote_values = list(dict.fromkeys(vote_values))
+            if initial_length != len(vote_values):
+                abort(400, 'Bad Request')
+
+        elif len(btl_vote) >= 12:
+            # Validate below the line votes
+            # Check if candidates exist
+            for candidate in btl_vote:
+                if not candidate['candidate_id'] in candidates_id_list:
+                    abort(400, 'Bad Request')
+
+            # Check if no duplicate values
+            initial_length = len(btl_vote)
+            vote_values = list(map(lambda vote: vote['number'], btl_vote))
+            vote_values = list(dict.fromkeys(vote_values))
+            if initial_length != len(vote_values):
+                abort(400, 'Bad Request')
+
+        else:
+            abort(400, 'Bad Request')
+    except:
+        abort(400, 'Bad Request')
+
+    # Create vote_data
+    db.execute('INSERT INTO vote_data (vote_data_blob, v_event_id) VALUES (?, ?)',
+                  (json.dumps(vote_data), v_event_id)
+    )
+
+    # Create vote_status
+    db.execute('INSERT INTO vote_status (user_id, v_event_id) VALUES (?, ?)',
+                  (user_id, v_event_id)
+    )
+
+    db.commit()
+
+    return '', 204
+
+# THIS ENDPOINT IS TO CHECK IF CREATE WORKS, DELETE LATER
+@bp.route('', methods=['GET'])
+@jwt_required
+def get():
+    # abort(403, 'Forbidden')

```

Comment:
This shouldn't be in production code when we "complete" the project. Just leaving this here for testing.

Reply:
Same with the two other endpoints below.